raven/backend.go

```diff
@@ -127,15 +119,27 @@ func fromGlogEvent(e glog.Event) *Event {
 			for k, v := range t {
 				data[k] = v
 			}
+		case glog.ErrorArg:
+			// Prepend the Message with the innermost error message.
+			// This causes it to be used for the headline.
+			eve.Message = t.RootCause().Error() + "\n\n" + message
```

Comment:
I foresee lots of 'context canceled' and similar headlines -- where the innermost is a generic library error and the "best" string is a level or two up. But I can't think of a general way to catch this. If we were using modules, would it be be possible to do "innermost error whose frame is in the same module as main"? Is that an API that exists? Anyway, irrelevant since we are not.

Definitely a big improvement over status quo in any case.

Reply:
Yeah, good point. Maybe the innermost frame that's not in the stdlib? I'm not sure that's easy to determine without hardcoding though...

Reply:
Maybe the innermost 2 frames is the best heuristic since it captures both the root error and the immediate caller's context?

Reply:
Yeah, innermost 2 seems like a good choice. Annoyingly it'd get less useful the more libraries were to adopt xerrors, which is backwards. But it probably works well in a majority of cases.